tests/test_foreman.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from urllib.parse import urlparse
+
+
+def test_foreman_version(api, variables):
+    if 'foreman_version' not in variables:
+        pytest.skip("'foreman_version' is not set")
```

Comment:
A more Python pattern is:
```python
try:
    expected_foreman_version = variables['expected_foreman_version']
except KeyError:
    pytest.skip("'expected_foreman_version' is not set")
else:
    status = api.resource('home').call('status')
    assert status['version'] == expected_foreman_version
```

Reply:
Can you educate me on what makes that more pythonic? when I google check if a key is in a dictionary I get the form I have.

Reply:
The Pythonic way is not to check if it is there, but rely on the `KeyError` that is raised if it's not.

Reply:
Indeed. https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html describes the style of checking as an anti-pattern. While your code is indeed the valid way of checking if a key is in the dict, it's unpythonic to even do so in the first place.

It's also easy to come up with an example where your example would be wrong.
```
from collections import defaultdict

data = defaultdict(lambda: 'a')
if 'x' in data:
    print(f"first in check: {data['x']}")
print(f"value: {data['x']}")
if 'x' in data:
    print(f"second in check: {data['x']}")
```

This outputs:
```
value: a
second in check: a
```

Note there is no `first in check: a` because at first the key is not present.

Reply:
That's still just telling me it is the way:

![4b2](https://user-images.githubusercontent.com/293363/102351006-c0996780-3f73-11eb-87b3-b1950b80fa2d.jpg)


Is there a guide/site/lint rule etc. you can point me to to learn about why its considered more Pythonic?

Reply:
There is no pylint yet in the repo, does it or some other tool catch and enforce this?

Reply:
It states:
> The Python community uses an EAFP (easier to ask for forgiveness than permission) coding style.

That the Python community uses it, makes it Pythonic. I don't think there's a tool to catch this because it's nearly impossible to catch it without introducing tons of false positives.

https://stackoverflow.com/questions/11360858/what-is-the-eafp-principle-in-python is another that says the same thing and refers to https://docs.python.org/3/glossary.html#term-eafp. That's probably the best source telling you that's the Python way.

Reply:
Thank y'all for the info! I gotta go unlearn some of my bad Python habits.